lib/rewrite/rewrite-expr-grammar.ym

```diff
@@ -120,16 +120,18 @@ rewrite_expr
             last_rewrite = log_rewrite_rename_new(configuration, from, to);
           }
           rewrite_rename_opts ')'              { $$ = last_rewrite; }
-	| KW_SET_TAG '(' string 
+	| KW_SET_TAG '(' template_content
 	  { 
 	    last_rewrite = log_rewrite_set_tag_new($3, TRUE, configuration);
-            free($3);
+            last_template_options = log_rewrite_set_get_template_options(last_rewrite);
+            log_template_unref($3);
           }
 	  rewrite_settag_opts ')'             	{ $$ = last_rewrite; }
-	| KW_CLEAR_TAG '(' string
+	| KW_CLEAR_TAG '(' template_content
           {
             last_rewrite = log_rewrite_set_tag_new($3, FALSE, configuration);
-            free($3);
+            last_template_options = log_rewrite_set_get_template_options(last_rewrite);
+            log_template_unref($3);
```

Comment:
If the template is a literal string, as we do not `ref()` it in `new()`, it will be `free()`d here, making `last_template_options` invalid.

Reply:
It won't invalidate that. Template options is embedded in the rewrite rule.

Reply:
Ah, I see. I mixed up `template` and `template_options`.